### ISSUE TITLE
[OIDC] Issuer URL wrong

### ISSUE THREAD

**Issue by srenatus** [cloudfoundry/uaa#652]
I believe the issue described in #425 has been not resolved:

- The seconds argument passed to `OpenIdConfiguration` is [_supposed to be the issuer_](https://github.com/cloudfoundry/uaa/blob/65a7d1ac2fefefb088de89a68633406ad9f1d3bf/model/src/main/java/org/cloudfoundry/identity/uaa/account/OpenIdConfiguration.java#L60-L61)...
- but what gets passed is [getTokenEndpoint](https://github.com/cloudfoundry/uaa/blob/65a7d1ac2fefefb088de89a68633406ad9f1d3bf/server/src/main/java/org/cloudfoundry/identity/uaa/account/OpenIdConnectEndpoints.java#L20)...
- leading to the wrong discovery data:
  ```
  $ curl http://localhost:8080/uaa/.well-known/openid-configuration | jq .issuer
  "http://localhost:8080/uaa/oauth/token"
  ```
- leading to issues with off-the-shelf oidc clients:
  ```
  $ bin/example-app --issuer http://localhost:8080/uaa
  error: Failed to query provider "http://localhost:8080/uaa": oidc: issuer did not match the issuer returned by provider, expected "http://localhost:8080/uaa" got "http://localhost:8080/uaa/oauth/token"
  ```

...or am I missing some configuration setting? I haven't changed the example config in any relevant way, I think.

Thanks a lot
Stephan

**Comment by bobhlo** [comment]
Is there an ETA for the fix - https://github.com/cloudfoundry/uaa/blob/master/server/src/main/java/org/cloudfoundry/identity/uaa/account/OpenIdConnectEndpoints.java#L20? 

The same issue also blocked us from creating UAA as OIDC identity providers in AWS ECR and DCOS.

**Comment by bsekar** [comment]
As per the [OIDC discovery spec](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationValidation), ```The issuer value returned MUST be identical to the Issuer URL that was directly used to retrieve the configuration information. This MUST also be identical to the iss Claim value in ID Tokens issued from this Issuer```. 
So, in case of UAA the `issuer` must not contain `/oauth/token`

**Comment by bsekar** [comment]
This has been fixed by #672
